interview-questions.md

```diff
@@ -8,42 +8,42 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: when your coding revolves around using objects. the data is stored in objects. functional programming is coding based around functions.
 
-Researched answer:
+Researched answer: In functional programming, data cannot be stored in objects and it can only be transformed by creating functions. In object-oriented programming, data is stored in objects.
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: floats are decimal numbers and integers are whole numbers.
 
-Researched answer:
+Researched answer: floats are decimal numbers and integers are whole numbers. its suggested to use the float data type in a database as opposed to integers because you can get more accurate values. 
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: not sure
 
-Researched answer:
+Researched answer: the final statement inside the method.
 
 4. What is a block in Ruby?
 
-Your answer:
+Your answer: chucks of code 
 
-Researched answer:
+Researched answer: ruby blocks are anonymous functions that can be passed into methods. blocks are enclosed in a "do" / "end" statement or between brackets {}. they can also have multiple arguments. 
 
 5. How do you extract a single value from a Ruby hash?
 
-Your answer:
+Your answer: using .extract! and .values
 
-Researched answer:
+Researched answer: using value methods.
```

Comment:
The question is asking about extracting a single value from a hash. Using .values will return an array of all of the values in the hash.